graphstorm-processing/tests/test_dist_executor.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import shutil
+import tempfile
+
+import pytest
+
+from graphstorm_processing.distributed_executor import DistributedExecutor, ExecutorConfig
+from graphstorm_processing.constants import ExecutionEnv, FilesystemType
+
+_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+@pytest.fixture(autouse=True, name="tempdir")
+def tempdir_fixture():
+    """Create temp dir for output files"""
+    tempdirectory = tempfile.mkdtemp(
+        prefix=os.path.join(_ROOT, "resources/test_output/"),
+    )
+    yield tempdirectory
+    shutil.rmtree(tempdirectory)
+
+
+def test_merge_input_and_transform_dicts(tempdir: str):
+    """Test run function with local data"""
+    input_path = os.path.join(_ROOT, "resources/small_heterogeneous_graph")
+    executor_configuration = ExecutorConfig(
+        local_config_path=input_path,
+        local_metadata_output_path=tempdir,
+        input_prefix=input_path,
+        output_prefix=tempdir,
+        num_output_files=-1,
+        config_filename="gsprocessing-config.json",
+        execution_env=ExecutionEnv.LOCAL,
+        filesystem_type=FilesystemType.LOCAL,
+        add_reverse_edges=True,
+        graph_name="small_heterogeneous_graph",
+        do_repartition=True,
+    )
+
+    dist_executor = DistributedExecutor(executor_configuration)
+
+    pre_comp_transormations = {
+        "node_features": {
+            "user": {
+                "age": {
+                    "transformation": "numerical",
+                }
+            }
+        },
+        "edge_features": {},
+    }
+
+    input_config_with_transforms = dist_executor._merge_config_with_transformations(
+        dist_executor.gsp_config_dict,
+        pre_comp_transormations,
+    )
+
+    # Ensure the "user" node type's "age" feature includes a transformation entry
```

Comment:
Does "age" feature has a transformation entry in is original config?
What if the "transformation" has other information like "normalizer": "min-max"?

Reply:
Yes, here we're just ensuring the `_merge_config_with_transformations` function works as intended, not actually checking the correctness of the transformation.

The contract we have for a transformation representation in the JSON file is that it always has one "transformation" key (or maybe we should call it "transformation_name") and the rest of the keys allow us to reconstruct the transformation.

For a numerical feature yes, we might have other keys like min_val/max_val, but this test would still apply (since we hardcode the representation).

In the follow-up PR we actually test the produced JSON representations and re-apply them to the data to ensure they match the original.

Reply:
I changed this test to use the categorical transformation to ensure we're using a feature for which we support transformations.

We test the actual saved values in `graphstorm-processing/tests/test_dist_category_transformation:test_multiple_single_cat_cols_json`